### PR TITLE
Document apiVersion 0.0.7, specVersion 0.0.5 and receipts in mappings

### PR DESCRIPTION
**Pull request by tilacog**
This PR briefly documents the changes introduced in https://github.com/graphprotocol/graph-node/pull/3373.

I'm not sure if those changes are enough for it to be considered to be final, so I'll be happy to refine this PR based on the team's suggestions.

The feature PR linked above contains lots of extra detail if more context is needed.

### CODE REVIEW

File: pages/en/developer/create-subgraph-hosted.mdx [hunk 1]
```diff
@@ -812,6 +812,22 @@ eventHandlers:
 
 An event will only be triggered when both the signature and topic 0 match. By default, `topic0` is equal to the hash of the event signature.
 
+## Transaction Receipts in Event Handlers
+
+Starting from `specVersion` `0.0.5` and `apiVersion` `0.0.7`, event handlers can have access to the receipt for the transaction which emitted them.
+
+To do so, event handlers must be declared in the subgraph manifest with thenew `receipt: true` key, which is optional and defaults to false.
```
Comment (evaporei) [inline pages/en/developer/create-subgraph-hosted.mdx:819]: Typo, `thenew` should have a space.

**Comment by evaporei** [comment]
Just warning for the reviewers, this shouldn't be merged until this is in the Hosted Service, for now it's only on the `master` branch of `graph-node`.

File: pages/en/developer/create-subgraph-hosted.mdx (hunk 1 above)
Comment (tilacog) [reply pages/en/developer/create-subgraph-hosted.mdx]: Thanks, fixed in b3e4456a

**Review by evaporei — APPROVED** [review]

**Review by azf20 — APPROVED** [review]
LGTM, excited for this one!

File: pages/en/developer/assemblyscript-api.mdx [hunk 2]
```diff
@@ -357,6 +358,7 @@ class Event {
   block: Block
   transaction: Transaction
   parameters: Array<EventParam>
+  receipt: TransactionReceipt | null
```
Comment (azf20) [inline pages/en/developer/assemblyscript-api.mdx:361]: @tilacog I think we need to add the schema for the TransactionReceipt object here so folks know what they can use? I don't think it is documented yet

**Comment by tuler** [comment]
Is there a performance penalty of defining `receipt: true`?

**Comment by azf20** [comment]
hey @tuler it requires an additional RPC call, so there will be an impact, but we have not yet got metrics for that

**Comment by ammanpashasc** [comment]
Hey, is this available yet for the hosted service?

**Comment by azf20** [comment]
Hey @ammanpashasc yes it is, are you having trouble with it?

**Comment by ammanpashasc** [comment]
> Hey @ammanpashasc yes it is, are you having trouble with it?

Hey, yeah so I've upgraded to graph-cli@0.29.2 and also updated the specVersion to 0.0.5 and apiVersion to 0.0.7. But when I try to add the `receipt: true` to event mapping, it gives me error: 

```
Path: dataSources > 0 > mapping > eventHandlers > 0
Unexpected key in map: receipt 
 ```
 
 This is what my  `.yaml` file looks like:
 
```
   - kind: ethereum/contract
    ......
      eventHandlers:
        - event: Assign(indexed address,uint256)
          handler: handleAssign
          receipt: true
      file: ./src/mappings/mapping.ts
```

Any idea what I'm doing wrong?

**Comment by benber86** [comment]
Does this also work for templates or is it only available for `dataSources`?
Using `graph-cli@0.30.3` and the hosted service, I've added `receipt: true` to my templates' `eventHandlers`:
```
      eventHandlers:
        - event: TokenExchange(indexed address,int128,uint256,int128,uint256)
          handler: handleTokenExchange
          receipt: true
```          
I then access `gasUsed` from the event's receipt:
```
export function handleTokenExchangeUnderlying(event: TokenExchangeUnderlying): void {
  const trade = event.params
  const receipt = event.receipt
  const gasUsed = receipt ? receipt.gasUsed : BIG_INT_ZERO 
```
But `receipt.gasUsed` is always null.

**Pull request merged**